packages/dd-trace/src/plugins/util/web.js

```diff
@@ -221,6 +221,20 @@ const web = {
       const context = contexts.get(req)
       context.error = context.error || error
     }
+  },
+
+  setUser (req, user) {
+    if (req || !user || !user.id) return false
+    
+    span = web.root(req)
+
+    if (!span) return false
+
+    for (const [k, v] of Object.entries(user)) {
+      span.setTag(`usr.${k}`, '' + v)
+    }
```

Comment:
How would you implement a more "guided" API anyway ?
Since the only really "strict" field is user.id, I'd rather have the rest be mostly transparent as requirements can change sometimes, but the public API should change at least as possible. wdyt ?